### ISSUE TITLE
Error while trying to set code style configuration from checkstyle xml

### ISSUE THREAD

**Issue by gabrielrolfsen** [jshiell/checkstyle-idea#387]
I am currently using the latest version of IntellIJ (2017.3.4) and the latest version of the plugin (5.18.2) and I am getting this error while trying to import settings from a checstyle xml configuration (Code Style -> Java > Import Scheme > Checkstyle configuration):

`CheckStyle Configuration import failed with error message: org.infernus.idea.checkstyle.exception.CheckstyleServiceException: Error executing command 'OpPeruseConfiguration': null`

The file is quite simple also:

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>
    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="FileTabCharacter">
        <property name="eachLine" value="true"/>
    </module>

    <module name="TreeWalker">

        <module name="OuterTypeFilename"/>

        <module name="EmptyBlock"/>

        <module name="NeedBraces"/>

        <module name="EmptyLineSeparator">
            <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
            <property name="tokens" value="METHOD_DEF"/>
        </module>

        <module name="AvoidStarImport" />

        <module name="CustomImportOrder">
            <property name="sortImportsInGroupAlphabetically" value="true"/>
        </module>

        <module name="RedundantImport"/>

    </module>
</module>

**Comment by jshiell** [comment]
Thanks for the report. I presume you've missed the top level `<module name="Checker">` on that snippet? In any case, I still get the error after adding it.

The full exception (from the `idea.log`) is:
```
2018-03-12 18:22:27,771 [ 287638]   WARN - er.CheckStyleCodeStyleImporter - Failed to import style 
org.infernus.idea.checkstyle.exception.CheckstyleServiceException: Error executing command 'OpPeruseConfiguration': null
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:147)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.peruseConfiguration(CheckstyleActionsImpl.java:123)
	at org.infernus.idea.checkstyle.importer.CheckStyleCodeStyleImporter.importConfiguration(CheckStyleCodeStyleImporter.java:93)
	at org.infernus.idea.checkstyle.importer.CheckStyleCodeStyleImporter.importScheme(CheckStyleCodeStyleImporter.java:56)
	at org.infernus.idea.checkstyle.importer.CheckStyleCodeStyleImporter.importScheme(CheckStyleCodeStyleImporter.java:22)
	at com.intellij.application.options.codeStyle.CodeStyleSchemesActions.a(CodeStyleSchemesActions.java:129)
	at com.intellij.application.options.codeStyle.CodeStyleSchemesActions.a(CodeStyleSchemesActions.java:100)
	at com.intellij.application.options.codeStyle.CodeStyleSchemesActions.importScheme(CodeStyleSchemesActions.java:63)
	at com.intellij.application.options.schemes.AbstractSchemeActions$ImportAction.actionPerformed(AbstractSchemeActions.java:309)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:220)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:237)
	at com.intellij.ui.popup.PopupFactoryImpl$ActionPopupStep.performAction(PopupFactoryImpl.java:966)
	at com.intellij.ui.popup.PopupFactoryImpl$ActionPopupStep.performAction(PopupFactoryImpl.java:956)
	at com.intellij.ui.popup.PopupFactoryImpl$ActionPopupStep.a(PopupFactoryImpl.java:950)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.ui.popup.AbstractPopup.a(AbstractPopup.java:1431)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.j(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:650)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:366)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.awt.Dialog.show(Dialog.java:1084)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:734)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:456)
	at com.intellij.openapi.ui.DialogWrapper.invokeShow(DialogWrapper.java:1696)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1645)
	at com.intellij.openapi.options.newEditor.SettingsDialog.a(SettingsDialog.java:83)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransactionAndWait(TransactionGuardImpl.java:153)
	at com.intellij.openapi.options.newEditor.SettingsDialog.show(SettingsDialog.java:83)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:97)
	at com.intellij.ide.actions.ShowSettingsAction.perform(ShowSettingsAction.java:63)
	at com.intellij.ide.MacOSApplicationProvider$Worker.a(MacOSApplicationProvider.java:118)
	at com.intellij.ide.MacOSApplicationProvider$Worker.a(MacOSApplicationProvider.java:212)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:74)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:424)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:407)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.j(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:650)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:366)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.NullPointerException
	at org.infernus.idea.checkstyle.importer.modules.CustomImportOrderImporter.createCustomImportTable(CustomImportOrderImporter.java:81)
	at org.infernus.idea.checkstyle.importer.modules.CustomImportOrderImporter.importTo(CustomImportOrderImporter.java:73)
	at org.infernus.idea.checkstyle.importer.CheckStyleCodeStyleImporter.lambda$importConfiguration$0(CheckStyleCodeStyleImporter.java:101)
	at org.infernus.idea.checkstyle.service.cmd.OpPeruseConfiguration.runVisitor(OpPeruseConfiguration.java:56)
	at org.infernus.idea.checkstyle.service.cmd.OpPeruseConfiguration.runVisitor(OpPeruseConfiguration.java:59)
	at org.infernus.idea.checkstyle.service.cmd.OpPeruseConfiguration.runVisitor(OpPeruseConfiguration.java:59)
	at org.infernus.idea.checkstyle.service.cmd.OpPeruseConfiguration.execute(OpPeruseConfiguration.java:45)
	at org.infernus.idea.checkstyle.service.cmd.OpPeruseConfiguration.execute(OpPeruseConfiguration.java:26)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:130)
	... 68 more
```

I'll have a quick look now but if it's involved then it may have to wait for the weekend.

**Comment by jshiell** [comment]
Looks like you're missing `customImportOrderRules` in your `CustomImportOrder` module. However, that's optional, so it looks like a faulty assumption in the import rule.

**Comment by jshiell** [comment]
Easy fix - done in 5.18.3. Any problems, please scream 😄

**Comment by mmoayyed** [comment]
Somewhat related issue: https://github.com/jshiell/checkstyle-idea/issues/377#issuecomment-404747579
